### ISSUE TITLE
Cloning fields between locales

### ISSUE THREAD

**Issue by zachshallbetter** [benjamminf/craft-neo#77]
Content from the default locale (EN) does not copy over to newly created locales (CN). This requires us to recreate content from scratch for every language and entry.

**Steps to reproduce:**
Add content to English locale.
Save and continue editing.
Click previously unselected locale.

<img width="1505" alt="screenshot 2017-01-18 15 42 20" src="https://cloud.githubusercontent.com/assets/949434/22087779/cc88ba86-dd94-11e6-845f-3efd413c8f05.png">

<img width="1500" alt="screenshot 2017-01-18 15 42 33" src="https://cloud.githubusercontent.com/assets/949434/22087784/d4135fd6-dd94-11e6-812e-284309a05afd.png">

**Comment by mrksmts** [comment]
Same goes for when you remove a locale and copy all the data to another or new locale.
All neo data is not being copied to the new locale.

**Comment by benjamminf** [comment]
Been something I've been working on for a while now – turns out locales are tricky 😅 I'll keep this thread updated instead of the other.

**Comment by benjamminf** [comment]
As I observed with #79, Matrix fields seem to behave in the same way as Neo in this instance – their blocks also do not get copied over. I actually thought they did, but it appears, at least for me, they do not.

Now the question becomes – should Neo steer away from the way Matrix behaves? Honestly, in this circumstance I feel like it should.

**Comment by zachshallbetter** [comment]
Yeah, I agree. Given users are using neo for every type of fields, I would say they should translation between all locales. I'm still unable to wrap my head around their thinking regarding not copying matrix data when "allow translation" is enabled.

**Comment by benjamminf** [comment]
It's pretty strange.

This functionality is pretty tricky to implement, and I've been sitting on this new release for far too long, so I'm going to be publishing a 1.4.1 today without having solved this one. If you can hold on a little longer I'll fix this after for a 1.4.2 release soon.

**Comment by zachshallbetter** [comment]
Awesome. Our team appreciates all of your hard work and attention.

**Comment by benjamminf** [comment]
I'm a little hesitant to implement this as there's so many edge and corner cases, as well as a lot of ambiguity as to whether content should be copied over or not.

Good news is, I'm working on a new feature that should get around this problem. I'm implementing a "copy/paste" feature which will let you copy and paste blocks or the entire field between entries and even different locales. This would mean simply marking all the blocks, copying them, then pasting them into each locale.

I'll keep this thread updated about that feature.

**Comment by akukral** [comment]
Is there any way to affect this on older entries, instead of just new entries?

**Comment by websNL** [comment]
Hi,

Nice, that the fix for saving a new entry is working with copying over the Neo Blocks. This saved us a massive amount of time for translations.

But we noticed one thing that didn't work. If you have a Neo Block with a Matrix in it, the Matrix content isn't filled / copied over to the other locale.

Is it possible to quickfix this issue? So that Matrix content in a Neo Block also get's copied over to the other locales on saving a new entry.

**Comment by mateostabio** [comment]
+1

**Comment by websNL** [comment]
Geachte relatie,Welkom bij de servicedesk van Webs. Voor vragen, bugs en wijzigingen aan uw website of hosting pakket, kunt u gebruik maken van dit Servicedesk systeem.Nadat een project of service is opgeleverd zijn er vaak nog veel aanvullende vragen. Om efficiënt antwoord te kunnen geven op deze vragen werken we met de Webs Servicedesk. Jij kunt met een eigen account 24-uur per dag al je vragen en wensen insturen. Doordat je deze instuurt via de Servicedesk kunnen wij aan onze kant deze vragen efficiënt oppakken door ze direct bij de juiste mensen neer te leggen. De voordelen van de Webs Servicedesk- 24/7 bereikbaar- Vragen worden sneller opgepakt en beantwoord - Inzage in de status van je vraagHoe maak je gebruik van de Webs Servicedesk?Voordat je gebruik kunt maken van de Servicedesk is het belangrijk om eerst even je account te activeren. Dit kun je doen via deze link: https://websnl.teamwork.com/support/#resetpassword/282461:97a4ca50-b1a3-4f86-8bd5-cfd3398d3e29:404580. Nadat je account is geactiveerd kun je gebruik maken van deze dienst door een e-mail met je vragen te sturen naar support@webs.nl of via de website https://websnl.teamwork.com/support.Met vriendelijke groet,Servicedesk medewerkers van Webs040 2324611support@webs.nl

**Comment by websNL** [comment]
Geachte relatie,Welkom bij de servicedesk van Webs. Voor vragen, bugs en wijzigingen aan uw website of hosting pakket, kunt u gebruik maken van dit Servicedesk systeem.Nadat een project of service is opgeleverd zijn er vaak nog veel aanvullende vragen. Om efficiënt antwoord te kunnen geven op deze vragen werken we met de Webs Servicedesk. Jij kunt met een eigen account 24-uur per dag al je vragen en wensen insturen. Doordat je deze instuurt via de Servicedesk kunnen wij aan onze kant deze vragen efficiënt oppakken door ze direct bij de juiste mensen neer te leggen. De voordelen van de Webs Servicedesk- 24/7 bereikbaar- Vragen worden sneller opgepakt en beantwoord - Inzage in de status van je vraagHoe maak je gebruik van de Webs Servicedesk?Voordat je gebruik kunt maken van de Servicedesk is het belangrijk om eerst even je account te activeren. Dit kun je doen via deze link: https://websnl.teamwork.com/support/#resetpassword/282461:a3f7fb36-cd68-43db-9bbf-b900debf26b4:443687. Nadat je account is geactiveerd kun je gebruik maken van deze dienst door een e-mail met je vragen te sturen naar support@webs.nl of via de website https://websnl.teamwork.com/support.Met vriendelijke groet,Servicedesk medewerkers van Webs040 2324611support@webs.nl

**Comment by websNL** [comment]
Geachte relatie,Welkom bij de servicedesk van Webs. Voor vragen, bugs en wijzigingen aan uw website of hosting pakket, kunt u gebruik maken van dit Servicedesk systeem.Nadat een project of service is opgeleverd zijn er vaak nog veel aanvullende vragen. Om efficiënt antwoord te kunnen geven op deze vragen werken we met de Webs Servicedesk. Jij kunt met een eigen account 24-uur per dag al je vragen en wensen insturen. Doordat je deze instuurt via de Servicedesk kunnen wij aan onze kant deze vragen efficiënt oppakken door ze direct bij de juiste mensen neer te leggen. De voordelen van de Webs Servicedesk- 24/7 bereikbaar- Vragen worden sneller opgepakt en beantwoord - Inzage in de status van je vraagHoe maak je gebruik van de Webs Servicedesk?Voordat je gebruik kunt maken van de Servicedesk is het belangrijk om eerst even je account te activeren. Dit kun je doen via deze link: https://websnl.teamwork.com/support/#resetpassword/282461:b81e5bac-9df8-420a-8abd-51fbd9468bcf:443986. Nadat je account is geactiveerd kun je gebruik maken van deze dienst door een e-mail met je vragen te sturen naar support@webs.nl of via de website https://websnl.teamwork.com/support.Met vriendelijke groet,Servicedesk medewerkers van Webs040 2324611support@webs.nl

**Comment by websNL** [comment]
Hi,

I see that the saving a new entry copies over the Neo Blocks is in the latest release.
Does that also include Neo Blocks with a Matrix field in it?

Previous comment on this:
_If you have a Neo Block with a Matrix in it, the Matrix content isn't filled / copied over to the other locale._

**Comment by mateostabio** [comment]
You need to edit the Neo field settings, make sure the "translatable" checkmark is not checked for this Neo field. Go back to your page with the Neo block, make a modification to the page (like adding a space) and hit save. It will copy everything over to the other locales!
